spinegeneric/cli/params_checker.py

```diff
@@ -54,113 +53,101 @@ def main():
-                        " "
-                        + item.filename
-                        + ": Missing: "
-                        + ManufacturersModelName
-                        + "; Cannot check parameters."
-                    )
-        else:
-            logging.warning(
-                " "
-                + item.filename
-                + ": Missing Manufacturer in json sidecar; Cannot check parameters."
-            )
+        # Check that the json sidecar has the correct keys and values
+        if "Manufacturer" not in item.get_metadata():
+            logging.warning(f" {item.filename}: Missing 'Manufacturer' key in json sidecar; Cannot check parameters.")
+            continue
+        Manufacturer = item.get_metadata()["Manufacturer"]
+        if Manufacturer not in data.keys():
+            logging.warning(f" {item.filename}: Manufacturer '{Manufacturer}' not in list "
+                            f"of known manufacturers: {data.keys()}. Cannot check parameters.")
+            continue
+        ManufacturersModelName = item.get_metadata()["ManufacturersModelName"]
+        if ManufacturersModelName not in data[Manufacturer].keys():
+            logging.warning(f" {item.filename}: Model '{ManufacturersModelName}' not present in list of known "
+                            f"models for manufacturer '{Manufacturer}'. Cannot check parameters.")
+            continue
+
+        # Parse the file's contrast from its suffix (sans `.nii.gz`)
+        Contrast = (item.filename.split("_")[-1]).split(".")[0]
+        # In the case of MTS files, manufacturers won't just specify 'MTS'. Instead, 'MTon_MTS',
+        # 'MToff_MTS', 'T1w_MTS' etc. will be used. So, we need to parse the type of MTS from the filename,
+        # then convert it to the specific names expected by the 'manufacturer params' dictionary.
+        if Contrast == "MTS":
+            # TODO: The manufacturer param dicts specifically contains the key "T1w_MTS", but I can't
+            #       find a single file in `data-multi-subject` that is MTS + T1w. Instead, all I see are
+            #       'mt-off' and 'mt-on'. So, this new method for parsing filenames passes for
+            #       data-multi-subject, but may fail for "T1w_MTS" data, if such data even exists?
+            try:
+                # Try new method for renamed, BIDS-compliant 'data-multi-subject'
+                MTS_acq = item.filename.split('_')[-2]
+                Contrast = {'mt-off': "MToff_MTS", 'mt-on': "MTon_MTS"}[MTS_acq]
+            except KeyError:
+                # Fall back to the old method for backwards compatibility with older datasets
+                Contrast = item.filename.split("_acq-")[1].split(".")[0]
+
+        # Fetch the names of each available parameter for the given manufacturer + model
+        keys_contrast = data[Manufacturer][ManufacturersModelName][str(Contrast)].keys()
+
+        # Validate repetition time against manufacturer's specifications
```

Comment:
The specs are actually _about_ manufacturers, but _by_ the spine-generic protocol.
```suggestion
        # Validate repetition time against spine-generic specifications
```